crates/pallet-domains/src/staking.rs

```diff
@@ -672,8 +686,8 @@ pub(crate) fn do_deregister_operator<T: Config>(
         let operator = maybe_operator.as_mut().ok_or(Error::UnknownOperator)?;
 
         ensure!(
-            *operator.status::<T>(operator_id) == OperatorStatus::Registered,
-            Error::OperatorNotRegistered
+            operator.is_operator_registered_or_deactivated::<T>(operator_id),
+            Error::OperatorNotAllowed
```

Comment:
`OperatorNotAllowed` doesn't seem like a clear description. Something like `OperatorNotRegisterdOrDeactivated` is long but makes it much clearer.